packages/x-date-pickers/src/internals/models/fields.ts

```diff
@@ -18,4 +18,5 @@ export interface BaseFieldProps<
   format?: string;
   disabled?: boolean;
   ref?: React.Ref<HTMLDivElement>;
+  onOpen: (event: React.UIEvent) => void;
```

Comment:
We need to discuss the exact DX here:

1. Should the behavior be `onOpen` or `onToggle`?
2. Should it be named `onOpen` or `onOpenViews` or `onOpenParent` or something else?

Right now it's not used by the built-in fields, but it will allow to make this demo and the button demo a lot simpler.
And as discussed last week, I would love to migrate to this approach for the built-in fields as well at some point, to make the picker more agnostic.

Note that this prop is not part of `DateFieldProps` but only of `DateFieldInPickerProps`, when used alone, the field are not polluted by this prop and should not be.